CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time

veronikaslc opened the pull request:
[CARDS-1725](https://phenotips.atlassian.net/browse/CARDS-1725)

Commit:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time

marta- reviewed (changes requested):
All the changes proposed in this PR should be contained in the `downtime-warning-module`, I don't see a reason why this extension-based functionality should affect `data-entry` and `homepage`.

Commit:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
Refactored code to aggregate all files related to the module into the module

marta- commented on modules/data-entry/src/main/resources/SLING-INF/nodetypes/dataentry.cnd:
```diff
@@ -900,3 +900,28 @@
+  - toDate (date)
+
+  // Whether to show the warning banner.
+  - enabled (BOOLEAN) = 'false'
```
This node definition also needs to move into a separate `.cnd` file in the `downtime-warning-banner` module.

marta- commented on modules/downtime-warning-banner/src/main/frontend/src/DowntimeWarningConfiguration.jsx:
```diff
@@ -0,0 +1,202 @@
+    makeStyles
+} from '@material-ui/core';
+
+import QuestionnaireStyle from "./questionnaire/QuestionnaireStyle";
```
This import is not used.
```suggestion
```

marta- commented on modules/downtime-warning-banner/src/main/frontend/package.json:
```diff
@@ -0,0 +1,43 @@
+    "@material-ui/core": "4.11.2",
+    "@material-ui/icons": "4.11.2",
+    "react": "16.14.0",
+    "moment": "2.29.1"
```
Unused dependency (I'm not sure if it's the only one).
```suggestion
```

marta- commented on modules/downtime-warning-banner/src/main/frontend/src/DowntimeWarningConfiguration.jsx:
```diff
@@ -0,0 +1,202 @@
+  const dateFormat = "yyyy-MM-dd hh:mm";
+
+  // Fetch saved admin config settings
+  let getQuickSearchSettings = () => {
```
Wrong name
```suggestion
  let getDowntimeWarningSettings = () => {
```

Commit:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
Fixed refactoring artefacts

Commit:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
Fixed more refactoring artefacts

Commit:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
Fixed pom for refactoring artefacts

marta- commented on modules/downtime-warning-banner/src/main/resources/SLING-INF/nodetypes/dataentry.cnd:
```diff
@@ -0,0 +1,46 @@
+//
```
This file should not be called `dataentry.cnd`. Rename to something along the lines of `downtimeConfig.cnd`.

veronikaslc committed:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time 

Renamed cnd file

veronikaslc commented:
Still do not load banner gonfig page then moved to the self module.

marta- committed:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
Enable the downtime warning banner in the distribution by default

marta- committed:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
Update Sling-Initial-Content rules after rebase

marta- committed:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
Specify the node type definition and load the inicial downtime configuration

marta- committed:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
Rename the *PageStart extensions to avoid clashes with the demo-banner extensions

marta- committed:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
Styling adjustments

marta- committed:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
Renamed boolean variable using `is` prefix instead of `on` which is normally used for naming handler functions

sashaandjic reviewed (approved):
tested

sdumitriu committed:
CARDS-1725: When there is scheduled maintenance downtime, display a warning banner to inform the users ahead of time
No need to make this module a feature

sdumitriu reviewed (approved):
The banner is not displayed in the PROMS UI. Should it be?

marta- commented:
> The banner is not displayed in the PROMS UI. Should it be?

It should, but we don't have an extension point there yet. This is reported in [CARDS-1741](https://phenotips.atlassian.net/browse/CARDS-1741) - _Add a "PageStart" extension point to the patient UI_.
